examples/tutorials/self-paced-training/part-3_security_and_privacy/chapter-5_Privacy_In_Federated_Learning/05.2_differential_privacy/privacy_with_differential_privacy.ipynb

```diff
@@ -0,0 +1,365 @@
+    "  initial_model=Net(),\n",
+    ")\n",
+    "\n",
+    "# Define the controller and send to server\n",
+    "controller = FedAvg(\n",
+    "    num_clients=n_clients,\n",
+    "    num_rounds=3,  # 100 rounds should converge\n",
+    ")\n",
+    "job.to_server(controller)\n",
+    "\n",
+    "# Add clients\n",
+    "for i in range(n_clients):\n",
+    "    runner = ScriptRunner(\n",
+    "        script=\"src/cifar10_fl.py\",\n",
+    "        params_transfer_type=TransferType.DIFF\n",
+    "    )\n",
+    "    job.to(runner, f\"site-{i+1}\")\n",
+    "\n",
+    "    # add privacy filter.\n",
+    "    dp_filter = SVTPrivacy(fraction=0.9, epsilon=0.1, noise_var=0.1, gamma=1e-5)\n",
+    "    job.to(dp_filter, f\"site-{i+1}\", tasks=[\"train\"], filter_type=FilterType.TASK_RESULT)\n",
+    "\n",
+    "# Optionally export the configuration\n",
+    "job.export_job(\"job_configs\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a661c68e-6b7f-4215-93e3-d4fe55eb5e7e",
+   "metadata": {},
+   "source": [
+    "Next, start the training"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "fc6b911d-a171-49b1-ad2e-b0d73032110c",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "job.simulator_run(f\"/tmp/nvflare/{job.name}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fcf57a13-842b-46e0-9032-69d3a03b3182",
+   "metadata": {},
+   "source": [
+    "> **Note:** you can also try adding or combining the filters with other privacy filters or customize them. For example, use the [PercentilePrivacy](https://nvflare.readthedocs.io/en/main/apidocs/nvflare.app_common.filters.html#nvflare.app_common.filters.PercentilePrivacy) filter based on Shokri and Shmatikov ([Privacy-preserving deep learning, CCS '15](https://dl.acm.org/doi/abs/10.1145/2810103.2813687)) or [ExcludeVars](https://nvflare.readthedocs.io/en/main/apidocs/nvflare.app_common.filters.html#nvflare.app_common.filters.ExcludeVars) filter to exclude variables that shouldn't be shared with the server."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e90c423d-4784-4956-961f-e2ea1ef1b30e",
+   "metadata": {},
+   "source": [
+    "### 7. Visualize the results\n",
+    "You can plot the results by running `tensorboard --logdir /tmp/nvflare` in a new terminal. In this notebook, we only run for a few FL rounds for simplicity. If you uncomment the recommended number of in the FedAvg controller definitions of the cells, you can run the experiments until convergence. As one can observe, the model with DP (red) takes more rounds to achieve a comparable training performance but has less risks of leaking private information compared to the model trained without DP (orange). For more details, on how to apply this filter in a medical imaging use case, see [Li et al. 2019](https://arxiv.org/abs/1910.00962) [1].\n",
+    "\n",
+    "![TensorBoard Training curve of FedAvg without and with DP](tb_curve_svt.png)"
```

Comment:
might be good to add a link to the next notebook after each notebook ends

Reply:
Good idea. Added to all notebooks in this chapter.